core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -171,7 +171,13 @@ public void onUpdate(List<ResolvedServerInfo> servers, Attributes config) {
         if (servers.isEmpty()) {
           onError(Status.UNAVAILABLE.withDescription("NameResolver returned an empty list"));
         } else {
-          loadBalancer.handleResolvedAddresses(servers, config);
+          try {
+            loadBalancer.handleResolvedAddresses(servers, config);
+          } catch (RuntimeException e) {
+            // It must be a bug! Push the exception back to LoadBalancer in the hope that it may be
+            // propagated to the application.
+            onError(Status.INTERNAL.withCause(e));
```

Comment:
The cause is often missing from Status's toString, and excludes most of the backtrace.  If the Status was printed out it would not be clear what caused it.  Maybe add "Unexpected Name Resolution failure"  or some other tidbit.